posthog/api/action.py

```diff
@@ -282,9 +281,29 @@ def _serialize_action(self, action: Action, filters: Dict[Any, Any], request: re
             'breakdown': []
         }
         if request.GET.get('shown_as', 'Volume') == 'Volume':
-            append.update(self._aggregate_by_day(action=action, filters=filters, request=request))
+            filtered_events = Event.objects.filter_by_action(action)
+            append.update(self._aggregate_by_day(filtered_events=filtered_events, filters=filters, request=request))
         elif request.GET['shown_as'] == 'Stickiness':
-            append.update(self._stickiness(action=action, filters=filters, request=request))
+            filtered_events = Event.objects.filter_by_action(action, order_by=None)
+            append.update(self._stickiness(filtered_events=filtered_events, filters=filters, request=request))
+        return append
+
+    def _serialize_event(self, event: Dict[Any, Any], request: request.Request) -> Dict:
+        append = {
+            'action': {
+                'id': event['id'],
+                'name': event['id']
```

Comment:
Is this right? I think everywhere else we do pass the event name around, but I'm not sure I follow this one

Reply:
I adhered to the format the setfilters uses from trendslogic for the moment which is why it does appears differently on backend